Zend/zend_compile.c

```diff
@@ -5711,6 +5717,93 @@ static zend_bool zend_is_valid_default_value(zend_type type, zval *value)
 	return 0;
 }
 
+static void zend_compile_attribute(zval *v, zend_ast *ast) /* {{{ */
+{
+	ZEND_ASSERT(ast->kind == ZEND_AST_ATTRIBUTE);
+
+	array_init_size(v, 1 + (ast->child[1] ? zend_ast_get_list(ast->child[1])->children : 0));
```

Comment:
Could possibly also store the `lcname` in addition to that, which would make various bits more efficient (but maybe those aren't performance sensitive in the first place).

Reply:
Maybe we could even avoid the HashTable in this case and use a struct (hack) like this?

```c
typeder struct {
    zend_string *name;
    uint32_t argc;
    zval argv[1];
} zend_attribute;
```
This would avoid memory overhead for the HashTable fields and the bucket allocation.

Reply:
Sounds reasonable. I don't think memory is a huge concern here, but probably implementing the more efficient representation is barely more effort than using a hashtable, so why not.